### PR TITLE
Update Argument.js

### PR DESCRIPTION
**Pull request by perilstar**
Fixes for the Union and Product static methods.

### CODE REVIEW

File: src/struct/commands/arguments/Argument.js [hunk 1]
```diff
@@ -366,7 +366,8 @@ class Argument {
             return null;
         }
 
-        if (typeof type === 'function') {
```
Comment (1Computer1) [inline src/struct/commands/arguments/Argument.js:369]: Fix this spacing.

Comment (perilstar) [reply src/struct/commands/arguments/Argument.js:369]: I didn't even realise I did that! Should be all good now.

Comment (1Computer1) [reply src/struct/commands/arguments/Argument.js:369]: Actually, this isn't right, Don't remove the `typeof`.

Comment (perilstar) [reply src/struct/commands/arguments/Argument.js]: My apologies. I'm perplexed; I don't have any memory of deleting that.

**Pull request merged**